esx_service/utils/auth_api.py

```diff
@@ -749,7 +813,10 @@ def _tenant_vm_replace(name, vm_list):
 
 def check_datastore(datastore_name):
     """ Check datastore with given name is a valid datastore or not """
-    if datastore_name == auth_data_const.DEFAULT_DS:
+    if datastore_name == auth_data_const.VM_DS:
+        return None
+
+    if datastore_name == auth_data_const.ALL_DS:
         return None
 
     if not vmdk_utils.validate_datastore(datastore_name):
```

Comment:
Validate_datastore() checks if the name is in the in-memory datastore cache. That depends on what datastores are visible on this host. But, thats not impacting this change.